src/Data/Term.hs

```diff
@@ -3,6 +3,12 @@ module Data.Term where
 import Control.Applicative ((<|>))
 import Data.Function (fix)
 import Data.List.NonEmpty (NonEmpty, last, unfoldr)
+import qualified Data.Map as Map
+import Data.Map (Map, findWithDefault, keys)
```

Comment:
I would like imports to be consistent, like this:
```haskell
import qualified Data.Map as Map
import Data.Map (Map)
```
I did not do so before because there were not that much data structures involved

Reply:
In addition, we do not need to keep the string keys ordered, so a `Data.HashMap` (and, correspondingly, `Data.HashSet`) would be a better fit.